resources/views/components/price.blade.php

```diff
@@ -0,0 +1,30 @@
+@props(['product' => 'product', 'type' => 'catalog', 'placeholder' => ''])
+
+<price :product="{{ $product }}" type="{{ $type }}">
+    <div
+        {{ $attributes->class('mt-1 flex items-center gap-2 text-14 font-semibold text-primary') }}
+        slot-scope="{ price, specialPrice, isDiscounted, range }"
+    >
+        <template v-if="range">
+            <span>
+                <template v-if="range.min != range.max">
+                    @{{ range.min | price }} - @{{ range.max | price }}
+                </template>
+                <template v-else>
+                    @{{ range.min | price }}
+                </template>
+            </span>
+        </template>
+        <template v-else>
+            <span v-text="$options.filters.price(specialPrice || price)">
```

Comment:
the specialPrice computed attribute already returns the price if no special price is entered.
(Which is also why it may be better to call it finalPrice)